### PR TITLE
feat: Add OT-3 State Manager (RQA-337)

### PR DESCRIPTION
**Pull request by DerekMaggio**
# Overview

Add OT-3 State Manager to `opentrons-emulation`

# Changelog

- Add `ot3-state-manager` target to `Dockerfile`
- Add `run-ot3-state-manager` option to entrypoint.sh
- Add concrete_ot3_service_builder.py
- Add `ot3-state-manager-exposed-port` to input file
- Add state manager env vars to all OT-3 emulators except for bootloader

# Review requests

None, @fsinapi  already tested against this branch, and @fsinapi doesn't make mistakes

# Risk assessment

Low

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -192,6 +192,11 @@ ENV OPENTRONS_HARDWARE "common-ot3-firmware"
 COPY entrypoint.sh /entrypoint.sh
 RUN /entrypoint.sh build
 
+FROM ot3-firmware-builder as ot3-state-manager
+ENV OPENTRONS_HARDWARE "ot3-state-manager"
+COPY entrypoint.sh /entrypoint.sh
+RUN /entrypoint.sh run
```
Comment (fsinapi) [inline docker/Dockerfile:198]: I had to change this to `ENTRYPOINT ["/entrypoint.sh", "run"]`

File: docker/entrypoint.sh [hunk 2]
```diff
@@ -129,6 +129,9 @@ case $FULL_COMMAND in
   run-ot3-gripper-hardware)
     /gripper-simulator
     ;;
+  run-ot3-state-manager)
+    (cd /ot3-firmware/state_manager && ../stm32-tools/poetry/bin/poetry run python3 -m state_manager.state_manager --right-pipette P1000-multi-96 localhost 8088)
```
Comment (fsinapi) [inline docker/entrypoint.sh:133]: The UDP server doesn't seem to receive anything if you pass in `localhost`. I had to change the last options here to `localhost 9999` to be able to send anything to/from the server.

File: docker/Dockerfile (hunk 1 above)
Comment (DerekMaggio) [reply docker/Dockerfile:198]: Yep, that totally makes sense. I was wondering why my build process was hanging indefinitely. Nice catch. Thank you!

Comment (DerekMaggio) [reply docker/Dockerfile]: 20b7b4008b5b2644cb2c76cefef7140c44ae19b3

File: docker/entrypoint.sh (hunk 2 above)
Comment (DerekMaggio) [reply docker/entrypoint.sh]: This makes sense as well. Thank you!

5eaa27df17a0b96abb8d747dd9f8a8d133393c51

**Review by fsinapi — APPROVED** [review]
LGTM! This branch lets me run an emulator and successfully communicate with the state manager.

**Pull request merged**